src/common/src/array/data_chunk.rs

```diff
@@ -289,6 +290,16 @@ impl DataChunk {
         column_idxes: &[usize],
         hasher_builder: H,
     ) -> Vec<HashCode> {
+        // If there is only one primary key column and the column is `SERIAL`,
+        // we skip the Hash process and extract the VnodeId directly
+        if let Ok(idx) = column_idxes.iter().exactly_one() &&
+            let ArrayImpl::Serial(array) = self.column_at(*idx).array_ref() {
+            return array
+                .iter()
+                .map(|item| item.map(|s| HashCode::from(extract_vnode_id_from_row_id(s.as_row_id()) as u64)).expect("SERIAL as RowId should not be None"))
+                .collect();
+        }
+
```

Comment:
It is not appropriate to modify the hash calculation code in this section. Our focus is solely on altering it for vnode computation purposes. It is worth noting that this hash value is utilized in various other sections, including hashAgg or other hash-based executors, and altering it can reduce the hash value range.

Reply:
Yes, it does need to be designed here, need a unified middle layer to abstract here, I'm still working on it

In fact, this draft pr is to test whether these modification can pass CI 🤣